yapapi/engine.py

```diff
@@ -107,6 +107,8 @@ def __init__(
         :param budget: maximum budget for payments
         :param strategy: market strategy used to select providers from the market
             (e.g. LeastExpensiveLinearPayuMS or DummyMS)
+        :param event_consumer: callable that will be directly executed on every Event this Engine creates.
+            NOTE: it is expected to be fast - if not, it will block the _Engine.
```

Comment:
And what if you want to add a consumer that may block?

Reply:
Then we create a non-blocking wrapper in a layer above the `_Engine`.
That's exactly what we're doing now with any consumer passed by the developer to `Golem.add_event_consumer`.